### PR TITLE
ROX-13025: RUN UI E2E test on PG

### PR DESCRIPTION
**Pull request by janisz**
https://github.com/stackrox/stackrox/pull/3369

### CODE REVIEW

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-nightlies-postsubmits.yaml [hunk 1]
```diff
@@ -417,6 +417,52 @@ postsubmits:
     - ^nightlies$
```
Comment (msugakov) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-nightlies-postsubmits.yaml:417]: Seems like you also need to make changes in
1. `ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml`
2. `ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-presubmits.yaml`

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-nightlies-postsubmits.yaml [hunk 2]
```diff
@@ -417,6 +417,52 @@ postsubmits:
...
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - ^nightlies$
+    cluster: build03
```
Comment (msugakov) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-nightlies-postsubmits.yaml:464]: Do you know, what does this `cluster:` attribute mean? It seems that the value is `build04` throughout this file, not `build03` you have here.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml [hunk 3]
```diff
@@ -145,6 +145,52 @@ postsubmits:
...
+    branches:
+    - ^master$
+    cluster: build04
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+      timeout: 5h0m0s
+    labels:
+      ci.openshift.io/generator: prowgen
+    max_concurrency: 1
```
Comment (msugakov) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml:159]:
```suggestion
    max_concurrency: 6
```
as @gavin-stackrox suggested.

Compare with the value for `branch-ci-stackrox-stackrox-master-merge-gke-ui-e2e-tests`.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-presubmits.yaml [hunk 4]
```diff
@@ -304,6 +304,57 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )gke-postgres-scale-tests,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (msugakov) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-presubmits.yaml:308]:
```suggestion
    always_run: false
```

as @gavin-stackrox suggested.
Compare with `pull-ci-stackrox-stackrox-master-gke-ui-e2e-tests`.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-presubmits.yaml [hunk 5]
```diff
@@ -304,6 +304,57 @@ presubmits:
...
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+      timeout: 5h0m0s
+    labels:
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-stackrox-stackrox-master-gke-postgres-ui-e2e-tests
+    optional: true
+    rerun_command: /test gke-postgres-ui-e2e-tests
```
Comment (msugakov) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-presubmits.yaml:323]:
```suggestion
    rerun_command: /test gke-postgres-ui-e2e-tests
    run_if_changed: ^ui/
```
would match what's in `pull-ci-stackrox-stackrox-master-gke-ui-e2e-tests`.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-nightlies-postsubmits.yaml [hunk 6]
```diff
@@ -228,6 +228,52 @@ postsubmits:
...
+    branches:
+    - ^nightlies$
+    cluster: build04
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+      timeout: 5h0m0s
+    labels:
+      ci.openshift.io/generator: prowgen
+    max_concurrency: 1
```
Comment (msugakov) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-nightlies-postsubmits.yaml:242]: @gavin-stackrox should `max_concurrency` be bumped here too, as well as in `branch-ci-stackrox-stackrox-nightlies-gke-ui-e2e-tests`?

**Review by msugakov — COMMENTED** [review]
@janisz can do diffs with `*gke-ui-e2e-tests` jobs yourself next time?

**Comment by janisz** [comment]
> @janisz can do diffs with *gke-ui-e2e-tests jobs yourself next time?

@msugakov I'm sorry, I did that but then need to rebase and it was changed. I should start from scratch instead of doing merge.

**Comment by msugakov** [comment]
What I mean is semantic diffs between "all things" `gke-ui-e2e-tests` and "all matching things" `gke-postgres-ui-e2e-tests`.

**Comment by msugakov** [comment]
Please merge https://github.com/stackrox/stackrox/pull/3369 first and make sure that `ci/rehearse/stackrox/stackrox/master/gke-postgres-ui-e2e-tests` job here succeeds.

**Comment by msugakov** [comment]
:facepalm: Did it merge automatically even despite `rehearse` job failing?

**Comment by janisz** [comment]
@msugakov Yes

**Pull request merged**